src/mediapipe_internal/mediapipegraphconfig.cpp

```diff
@@ -83,14 +83,14 @@ Status MediapipeGraphConfig::parseNode(const rapidjson::Value& v) {
         if (v.HasMember("base_path")) {
             std::string providedBasePath(v["base_path"].GetString());
             if (providedBasePath.size() == 0)
-                this->setBasePath(this->getGraphName() + "/");
-            else if (providedBasePath.back() == '/')
+                this->setBasePath(this->getGraphName() + std::string(1, std::filesystem::path::preferred_separator));
+            else if (providedBasePath.back() == std::string(1, std::filesystem::path::preferred_separator).back())
```

Comment:
Is this back() needed?
```suggestion
            else if (providedBasePath.back() == std::string(1, std::filesystem::path::preferred_separator))
```

Reply:
Yes, we need char to compare.